lib/rubocop/cop/style/case_condition_repeat.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks that there are no repeated conditions
+      # used in case 'when' expressions
+      #
+      # @example
+      #
+      #    # bad
+      #    case x:
+      #    when 'first'
+      #      do_something
+      #    when 'first'
+      #      do_something_else
+      #
+      # Reports a new offence for each conditional value that is repeated
+      class CaseConditionRepeat < Cop
```

Comment:
This is 90% a personal preference, but I'd probably guess the cop name to be something like `DuplicateCaseCondition`. 

I'm an opponent of pushing a word to the front of a constant name to "organize" it, as I think organization should be handled with language constructs (modules), and constant names should read like human. (I wouldn't walk over to my colleague and say: "You have a case condition repeat.") 😀